### PR TITLE
stub

### PR DESCRIPTION
**Pull request by TianGan-okta**

### CODE REVIEW

File: src/sinonSandbox.js [hunk 1]
```diff
@@ -1,15 +1,76 @@
-export function isSinonFunction(value, t) {
-    return t.isCallExpression(value.expression.right)
-        && value.expression.right.callee.object.name === 'sinon'
+export function removeSinonFunction(path, t) {
+    const value = path.node;
+    if (t.isCallExpression(value.expression.right)
+        && value.expression.right.callee.object.name === 'sinon') {
+        if (t.isMemberExpression(value.expression.left)
+            && value.expression.right.callee.property.name === 'createSandbox') {
+            path.findParent((path) => path.isProgram()).traverse({
```
Comment (michaeltamaki-okta) [inline src/sinonSandbox.js:7]: Good idea!

File: src/sinonSandbox.js [hunk 2]
```diff
@@ -1,15 +1,76 @@
...
+                    }
+                }
+            });
+            path.remove();
+        }
+    }
 }
 
-export function isRestoreSandbox(value, t) {
+export function isRestoreSandbox(value, t, name) {
```
Comment (michaeltamaki-okta) [inline src/sinonSandbox.js:45]: 👏

File: test/fixtures/sinon-stub/output.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  const obj = {
+    method: (num) => num + 10,
+    add: (a, b) => a + b,
+  };
+
+  function setupMock(ss) {
+    this.stubs = {
+      add: jest.spyOn(obj, "add").mockImplementation(() => {
+        $.Deferred().resolve();
+      }),
```
Comment (michaeltamaki-okta) [inline test/fixtures/sinon-stub/output.js]: Don't forget to return the deferred!
```suggestion
      add: jest.spyOn(obj, "add").mockImplementation(() => $.Deferred().resolve()),
```

**Review by michaeltamaki-okta — APPROVED** [review]

File: test/fixtures/sinon-stub/output.js (hunk 3 above)
Comment (TianGan-okta) [reply test/fixtures/sinon-stub/output.js]: Making it return statement so that if it returns a complex object, we can still handle

**Pull request merged**